google/resource_compute_ssl_policy.go

```diff
@@ -0,0 +1,244 @@
+			// CUSTOM profile demands non-zero custom_features, and other profiles (i.e., not CUSTOM) demand zero custom_features
+			if diff.HasChange("profile") || diff.HasChange("custom_features") {
+				if profile.(string) == "CUSTOM" {
+					if customFeaturesCount.(int) == 0 {
+						return fmt.Errorf("Error in SSL Policy %s: the profile is set to %s but no custom_features are set.", diff.Get("name"), profile.(string))
+					}
+				} else {
+					if customFeaturesCount != 0 {
+						return fmt.Errorf("Error in SSL Policy %s: the profile is set to %s but using custom_features requires the profile to be CUSTOM.", diff.Get("name"), profile.(string))
+					}
+				}
+				return nil
+			}
+			return nil
+		},
+	}
+}
+
+func resourceComputeSslPolicyCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	sslPolicy := &computeBeta.SslPolicy{
+		Name:           d.Get("name").(string),
+		Description:    d.Get("description").(string),
+		Profile:        d.Get("profile").(string),
+		MinTlsVersion:  d.Get("min_tls_version").(string),
+		CustomFeatures: convertStringSet(d.Get("custom_features").(*schema.Set)),
+	}
+
+	op, err := config.clientComputeBeta.SslPolicies.Insert(project, sslPolicy).Do()
+	if err != nil {
+		return fmt.Errorf("Error creating SSL Policy: %s", err)
+	}
+
+	d.SetId(sslPolicy.Name)
+
+	err = computeSharedOperationWaitTime(config.clientCompute, op, project, int(d.Timeout(schema.TimeoutCreate).Minutes()), "Creating SSL Policy")
+	if err != nil {
+		d.SetId("") // if insert fails, remove from state
+		return err
+	}
+
+	return resourceComputeSslPolicyRead(d, meta)
+}
+
+func resourceComputeSslPolicyRead(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	name := d.Id()
+
+	sslPolicy, err := config.clientComputeBeta.SslPolicies.Get(project, name).Do()
+	if err != nil {
+		return handleNotFoundError(err, d, fmt.Sprintf("SSL Policy %q", name))
+	}
+
+	d.Set("name", sslPolicy.Name)
+	d.Set("description", sslPolicy.Description)
+	d.Set("min_tls_version", sslPolicy.MinTlsVersion)
+	d.Set("profile", sslPolicy.Profile)
+	d.Set("fingerprint", sslPolicy.Fingerprint)
+	d.Set("project", project)
+	d.Set("self_link", ConvertSelfLinkToV1(sslPolicy.SelfLink))
+	d.Set("enabled_features", convertStringArrToInterface(sslPolicy.EnabledFeatures))
+	d.Set("custom_features", convertStringArrToInterface(sslPolicy.CustomFeatures))
+
+	d.SetId(sslPolicy.Name)
+
+	return nil
+}
+
+func resourceComputeSslPolicyUpdate(d *schema.ResourceData, meta interface{}) error {
+
+	config := meta.(*Config)
+
+	project, err := getProject(d, config)
+	if err != nil {
+		return err
+	}
+
+	name := d.Get("name").(string)
+
+	sslPolicy := &computeBeta.SslPolicy{
+		Fingerprint: d.Get("fingerprint").(string),
+	}
+
+	if v, ok := d.GetOk("profile"); ok {
```

Comment:
nit: There isn't actually any difference between doing it this way and just setting it in the struct, since if the value is empty it'll be the empty string in the struct regardless.